metadata/src/main/java/org/apache/kafka/controller/metrics/ControllerMetadataMetrics.java

```diff
@@ -39,10 +44,33 @@
  * {@link org.apache.kafka.controller.metrics.QuorumControllerMetrics}, not here.
  */
 public final class ControllerMetadataMetrics implements AutoCloseable {
+    @InterfaceStability.Stable
+    public enum BrokerRegistrationState {
```

Comment:
I think it would make sense to split this out into a separate file, since it's annotated as stable and `ControllerMetadataMetrics` is not.